packages/nimble-components/src/select/specs/IxD.md

```diff
@@ -0,0 +1,293 @@
+| `ENTER` or `SPACE`     | Opens the menu without moving focus or changing selection                                                        |
+| `DOWN ARROW`           | Opens the menu without moving focus or changing selection (DOM focus remains on the select)                      |
+| `ALT` + `DOWN ARROW`   | Opens the menu without moving focus or changing selection                                                        |
+| `UP ARROW`             | Opens the menu and moves visual focus to the first menu item (DOM focus remains on the select)                   |
+| `HOME`                 | Opens the menu and moves visual focus to the first menu item                                                     |
+| `END`                  | Opens the menu and moves visual focus to the last menu item                                                      |
+| _Printable characters_ | Opens the menu (if closed), moves visual focus to the first menu item that matches the typed character or string |
+
+`TAB` focuses the select component from the previous or next focusable element on the page.
+
+![ ](spec-images/select-26.png)
+
+`ENTER` or `SPACE` on the focused component opens the menu and maintains current focus and (if applicable) menu item selection.
+
+![ ](spec-images/select-27.png)
+
+`DOWN ARROW` on the focused component opens the menu and maintains current focus and (if applicable) menu item selection.
+
+![ ](spec-images/select-28.png)
+
+> NOTE: NEEDS DEV REVIEW & VISUAL DESIGN?
+
+`UP ARROW` on the focused component opens the menu and focuses (but does not select) the first or last menu item.
+
+![ ](spec-images/select-29.png)
+
+#### Menu
+
+> QUESTION: WHAT KEYBOARD SHORTCUT SHOULD BE USED TO FOCUS THE FIRST MENU ITEM AFTER THE FILTER BAR?
+
+| Key                        | Description                                                                                                 |
+| -------------------------- | ----------------------------------------------------------------------------------------------------------- |
+| `ENTER` or `SPACE`         | Sets the value of the field button to the focused menu item, closes the menu, and moves focus to the select |
+| `DOWN ARROW` or `UP ARROW` | Moves visual focus to the next or previous menu item                                                        |
+| `ESC`                      | Closes the menu and moves visual focus to the select                                                        |
+| `HOME` or `END`            | Moves visual focus to the first or last menu item                                                           |
+| `PAGEUP` or `PAGEDOWN`     | Jumps visual focus up or down ten menu items (or to first or last menu item)                                |
+| _Printable characters_     | Moves visual focus to the first menu item that matches the typed character or string                        |
```

Comment:
Is this how you want it to behave for a filterable select, versus opening the popup, and using the printable characters as the filter text?

Reply:
I'm not sure what you mean. The filter is in the popup.

Reply:
I read the line of "Moves visual focus to the first menu item that matches the typed character or string" as: 

"If a user has the `Select` focused and then types a printable character that it will open the popup and focus the first item in the dropdown that either starts with or contains (this part is unclear) the typed in text". 

This is different than if the typed in text just immediately being forwarded to the filter input in the dropdown, if the `Select` _is_ filterable.

Reply:
Ah, I get it. No, I think it would be best to forward the text to the filter input in the dropdown. I've added that detail.